packages/common-all/src/engine/EngineV3Base.ts

```diff
@@ -301,43 +304,39 @@ export abstract class EngineV3Base implements ReducedDEngine {
     };
   }
 
+  /**
+   * See {@link DEngine.queryNotes}
+   */
   async queryNotes(opts: QueryNotesOpts): Promise<QueryNotesResp> {
-    // const ctx = "Engine:queryNotes";
-    const { vault } = opts;
-    const MAX_LIMIT = 100;
-
-    // Need to ignore this because the engine stringifies this property, so the types are incorrect.
-    // @ts-ignore
-    if (vault?.selfContained === "true" || vault?.selfContained === "false")
-      vault.selfContained = vault.selfContained === "true";
-    const response = await this.noteStore.queryMetadata(opts);
+    const response = await this.noteStore.query(opts);
     if (response.isErr()) {
       // TODO: need to return an error
       return [];
     }
+
     let items = response.value;
-    if (items.length === 0) {
-      return [];
+    if (items.length > this.API_MAX_LIMIT) {
+      items = items.slice(0, this.API_MAX_LIMIT);
     }
+    return items;
+  }
 
-    if (!_.isUndefined(vault)) {
-      items = items.filter((ent) => {
-        return VaultUtils.isEqual(vault, ent.vault, this.wsRoot);
-      });
+  /**
+   * See {@link DEngine.queryNotesMeta}
+   */
+  async queryNotesMeta(opts: QueryNotesOpts): Promise<QueryNotesMetaResp> {
+    const response = await this.noteStore.queryMetadata(opts);
+    if (response.isErr()) {
+      // TODO: need to return an error
+      return [];
```

Comment:
can we throw here? then we can know when it goes wrong at least.
of course we can also change this to return `Promise<RespV3<QueryNotesMetaResp>>`